gulpfile.babel.js

```diff
@@ -0,0 +1,264 @@
+			markup: `${mainDirectories.dist}`,
+			styles: `${mainDirectories.dist}assets/css/`,
+			scripts: `${mainDirectories.dist}app/`,
+			images: `${mainDirectories.dist}assets/img/`,
+			libs: `${mainDirectories.dist}libs/`
+		}
+	},
+	autoPrefix: [
+		'> 1%',
+		'last 3 version',
+		'ie 8',
+		'ie 9',
+		'Firefox ESR',
+		'Opera 12.1'
+	]
+};
+
+function onError(err) {
+	notify({
+		title: 'Gulp Task Error',
+		subtitle: 'Plugin: <%= error.plugin %>',
+		message: 'Check the console.'
+	}).write(err);
+
+	console.log(err.toString());
+
+	this.emit('end');
+}
+
+export function clean() {
+	if (isProdBuild()) {
+		return del(['dist']);
+	}
+	return del(['server']);
+}
+
+export function styles() {
+	if (isProdBuild()) {
+		return gulp.src(settings.sources.stylesEntryPoint)
+			.pipe(plumber({errorHandler: onError}))
+			.pipe(less({
+				plugins: [new Autoprefix({
+					browsers: settings.autoPrefix
+				})]
+			}))
+			.pipe(cleanCss())
+			.pipe(rename({
+				baseName: 'index',
+				suffix: '.min'
+			}))
+			.pipe(gulp.dest(settings.destinations.prod.styles))
+			.pipe(rename('index.uncss.min.css'))
+			.pipe(uncss({
+				html: [settings.sources.markup]
+			}))
+			.pipe(gulp.dest(settings.destinations.prod.styles));
+	}
+	return gulp.src(settings.sources.stylesEntryPoint)
+		.pipe(plumber({errorHandler: onError}))
+		.pipe(sourcemaps.init())
+		.pipe(less({
+			plugins: [new Autoprefix({
+				browsers: settings.autoPrefix
+			})]
+		}))
+		.pipe(sourcemaps.write('./'))
+		.pipe(gulp.dest(settings.destinations.dev.styles));
+}
+
+export function images() {
+	if (isProdBuild()) {
+		return gulp.src(settings.sources.images)
+			.pipe(imagemin())
+			.pipe(gulp.dest(settings.destinations.prod.images));
+	}
+	return gulp.src(settings.sources.images)
+		.pipe(gulp.dest(settings.destinations.dev.images));
+}
+
+export function clientScripts() {
+	if (isProdBuild()) {
+		return gulp.src(settings.sources.scripts)
+			.pipe(rollup({
+				entry: './src/app/index.js'
+			}))
+			.pipe(babel())
+			.pipe(uglify())
+			.pipe(rename({
+				baseName: 'client',
+				suffix: '.min'
+			}))
+			.pipe(gulp.dest(settings.destinations.prod.scripts));
+	}
+	return gulp.src(settings.sources.scripts)
+		.pipe(rollup({
+			entry: './src/app/index.js'
+		}))
+		.pipe(babel())
+		.pipe(gulp.dest(settings.destinations.dev.scripts));
+}
+
+export function vendorScripts() {
+	const b = browserify(Object.assign({}, browserifyInc.args));
+	settings.sources.externalJs.forEach(dep => b.require(dep));
+	browserifyInc(b, {cacheFile: './.browserify-cache.json'});
+
+	if (isProdBuild()) {
+		return b.bundle()
+			.pipe(source('vendor.min.js'))
+			.pipe(buffer())
+			.pipe(uglify())
+			.pipe(gulp.dest(settings.destinations.prod.libs));
+	}
+	return b.bundle()
+		.pipe(source('vendor.js'))
+		.pipe(buffer())
+		.pipe(gulp.dest(settings.destinations.dev.libs));
+}
+
+export function bundleExternalCSS(done) {
+	const files = require('./package.json').bootstrapKickstart.bundleCSS;
```

Comment:
In the future we might reuse the package.json, so maybe import it in the head of the file and use it via reference.